android/beagle/src/test/java/br/com/zup/beagle/android/view/mapper/SendRequestActionMapperTest.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package br.com.zup.beagle.android.view.mapper
+
+import br.com.zup.beagle.android.BaseTest
+import br.com.zup.beagle.android.networking.ResponseData
+import org.junit.Test
+
+class SendRequestActionMapperTest : BaseTest(){
+
+    @Test
+    fun `GIVEN response data is an integer WHEN toResponse is called THEN it should return an Int object`() {
+        //Given
+        val responseData = ResponseData(
+            statusCode = 200,
+            data = "10".toByteArray()
+        )
+
+        //When
+        val response = SendRequestActionMapper.toResponse(responseData)
+
+        //Then
+        assert(response.data is Int)
+        assert(response.data == 10)
+    }
+
+    @Test
+    fun `GIVEN response data is an valid json WHEN toResponse is called THEN it should return an Any object`() {
+        //Given
+        val responseData = ResponseData(
+            statusCode = 200,
+            data = "{\"result\":\"test\"}".toByteArray()
+        )
+
+        //When
+        val response = SendRequestActionMapper.toResponse(responseData)
+
+        //Then
+        assert(response.data is Any)
```

Comment:
Maybe you could change this assert to:
```suggestion
       val jsonObject = response.data is JSONObject
        assertEquals("test", jsonObject.getString("result"))
```

Because the expected is to convert a JSON String to a valid context data value.
Maybe you should create another cases, like JSONArray, double, int, boolean.

Reply:
Or even create other test case for this suggested assertion, because the actual assertion make sense with THEN clause in test case.

Reply:
added new test cases

Reply:
It's missing only the JSONArray test and we can go.

Reply:
added